src/index/src/inverted_index/search/fst_apply/intersection_apply.rs

```diff
@@ -32,7 +35,41 @@ pub struct IntersectionFstApplier {
     ranges: Vec<Range>,
 
     /// A list of `Dfa` compiled from regular expression patterns.
-    dfas: Vec<DFA<Vec<u32>>>,
+    dfas: Vec<DfaFstAutomaton>,
+}
+
+struct DfaFstAutomaton(DFA<Vec<u32>>);
+
+impl fst::Automaton for DfaFstAutomaton {
+    type State = StateID;
+
+    #[inline]
+    fn start(&self) -> Self::State {
+        let config = Config::new();
+        self.0.start_state(&config).unwrap()
```

Comment:
I don't like this `unwrap`. But it's the API's signature of fst.

Hello @BurntSushi! I follow your suggestion in https://github.com/rust-lang/regex/issues/1087#issuecomment-1711601092 and the original impl at https://github.com/BurntSushi/regex-automata/blob/0.1.10/src/transducer.rs to wrap this implementation.

Since you're the author of all these crates (fst, regex-automata, regex), could you give a look on whether it's a proper implementation? Also if we can avoid such `unwrap`.